components/scream/src/physics/p3/p3_functions_f90.cpp

```diff
@@ -181,5 +189,73 @@ void access_lookup_table_coll_f(Int dumjj, Int dumii, Int dumj, Int dumi, Int in
   *proc = P3F::apply_table_coll(qiti_gt_small, adjusted_index, P3GlobalForFortran::itabcol(), ti, tr)[0];
 }
 
+void cloud_water_conservation_f(Real qc_, Real qcnuc_, Real dt, Real* qcaut_, Real* qcacc_, Real* qccol_,
+  Real* qcheti_, Real* qcshd_, Real* qiberg_, Real* qisub_, Real* qidep_)
+  {
+    using P3F = Functions<Real, HostDevice>;
+
+    typename P3F::Spack qc(qc_);
+    typename P3F::Spack qcnuc(qcnuc_);
```

Comment:
You can reuse the type to declare several values if you want.

```
typename P3F::Spack v1(v1_), v2(v2_) ...
```

Reply:
I'm fine with this either way. Do we want to set a standard for this since such code will appear time and again. I find using separate lines to be more readable. But if you think reducing the number of lines is better, I would be ok with that too.

Reply:
It's fine to leave as-is if you find it to be more readable.